### PR TITLE
Solves issue #7 API: Add new user #7 

### PR DESCRIPTION
**Pull request by prashanthreddykallem**
This commit solves issue https://github.com/mariogarridopt/bookit/issues/7. It validates the user input using Pydantic library. After validating it adds the input to the db and returns it to the user. It check for duplicates and sends a 409 error to the user. It raises 500 internal server if anything else goes wrong

### CODE REVIEW

**Comment by mariogarridopt** [comment]
Hey! Thanks for your contribution @prashanthreddykallem.

Before we merge the changes, there are some things I would like to discuss:
- The  `.vscode` on `/api/.gitignore` should not be required since it is already on `/.gitignore`, the one inside `/api` is just to ignore API (python) related stuff;
- Thanks for adding `schemas.schemas` but we already have them on `/api/models/auth.py` please use that one instead of creating new ones so that we don't have duplicated code;
- I don't think we need flask_pydantic, if you disagree please let me know why is required;
- You don't need to do an insert + a select, the insert method already returns the last inserted ID;
- As specified on the issue #7 if the "add" succeeds you should present a JSON with the user data and send a `201` status code;
- Please run the lint command (`docker exec bookit-api pylint "*"`) there are some lint errors that should be fixed'd before we merge.

Thanks for the patience, we will make this work great 😄

**Review by mariogarridopt — CHANGES_REQUESTED** [review]
Hey! Thanks for your contribution @prashanthreddykallem.

Before we merge the changes, there are some things I would like to discuss:
- The  `.vscode` on `/api/.gitignore` should not be required since it is already on `/.gitignore`, the one inside `/api` is just to ignore API (python) related stuff;
- Thanks for adding `schemas.schemas` but we already have them on `/api/models/auth.py` please use that one instead of creating new ones so that we don't have duplicated code;
- I don't think we need flask_pydantic, if you disagree please let me know why is required;
- You don't need to do an insert + a select, the insert method already returns the last inserted ID;
- As specified on the issue #7 if the "add" succeeds you should present a JSON with the user data and send a `201` status code;
- Please run the lint command (`docker exec bookit-api pylint "*"`) there are some lint errors that should be fixed'd before we merge.

Thanks for the patience, we will make this work great 😄

**Comment by prashanthreddykallem** [comment]
Hey Mario, 

Thank you for reviewing the pull request. 
I will make the required changes as mentioned in point 1 and 4. 
For point 2 and 3 :
Pydantic allows us to validate the input and output JSONs. Since the project is heavily based on API(s), using pydantic would make the validation part easier. For pydantic to work I would need a schemas file. I have created that in schemas/schemas.py. 
Let me know if you want to keep pydantic or not. I will make the changes accordingly. 

Thank you.

**Comment by mariogarridopt** [comment]
Sorry for the delay,

For a Pydantic implementation I think the best place to do it is at the Model level, or the BaseModel, those will serve as your schema, no need to create new files.

But I don't want to discuss that in this PR, I took few minutes to study Pydantic and we might use it in the future, I'm creating a new issue (#13) as a SPIKE to try to implement Pydantic in the hole project and see what it looks like. Please drop your expertise there, but, for this ticket I think we should implement it without Pydantic.

File: api/controllers/auth_controller.py [hunk 1]
```diff
@@ -77,10 +80,17 @@ def get_users_list(_) -> dict:
 
 @cross_origin()
 @admin_required
-def add_user(_) -> dict:
+@validate(body = AuthRequestModel)
```
Comment (mariogarridopt) [inline api/controllers/auth_controller.py:83]: Sync with `master`. Pydantics is now supported by the project, you can just change this to:
```suggestion
@validate(body = Auth)
```

All Schemas can be removed

**Comment by mariogarridopt** [comment]
Since the progress on this PR has slated, a fix was made here: #36
Issue #7 is now closed.

Thanks for your contribution

**Pull request closed without merging**